src/core/pratt.ml

```diff
@@ -0,0 +1,133 @@
+(** A simple parser for operators. See Pratt parsing or the Shunting Yard
+    algorithm.
+
+    <https://dev.to/jrop/pratt-parsing>
+    <https://effbot.org/zone/simple-top-down-parsing.htm> *)
+
+open Lplib
+open Lplib.Extra
+
+open Syntax
+
+(** Type of operators with additional data. *)
+(* REVIEW: simplify the type into [Unary of float * assoc] and remove [unop]
+   and [binop] *)
+type operator =
+  | Unary of unop
+  | Binary of binop
+
+(** Table containing all registered binary and unary operators that may appear
+    in terms parsed by {!val:Pratt.expression}. *)
+let operators : operator StrHtbl.t = StrHtbl.create 17
+
+module Pratt : sig
+  val expression : ?rbp:priority -> p_term Stream.t -> p_term
+  (** [expression rbp s] parses stream of tokens [s] with right binding power
+      [rbp]. It transforms a sequence of applications to a structured
+      application tree containing infix and prefix operators. For instance,
+      assuming that [+] is declared infix, it transforms [3 + 5 + 2],
+      represented as [@(@(@(@(3,+),5),+),2)] (where [@] is the application)
+      into [(@(+(@(+,3,5)),2)]. Note that it doesn't recurse into terms such
+      as abstractions or arrows. *)
+end = struct
+
+  (** [get_ident t] extracts a string and a position from [t] if it is an
+      identifier. *)
+  let get_ident : p_term -> (string * Pos.popt) option = fun {elt; pos} ->
+    match elt with
+    | P_Iden({elt=(_,s); _}, _) -> Some(s, pos)
+    | _ -> None
+
+  (** [lbp t] returns the left binding power of term [t] (which is 0 if [t] is
+      not an operator). *)
+  let lbp : p_term -> priority = fun pt ->
+    match get_ident pt with
+    | Some(s,_) ->
+        begin match StrHtbl.find_opt operators s with
+          | Some (Binary(_, _, bp, _))
+          | Some (Unary(_, bp, _)) -> bp
+          | None -> assert false
+        end
+    | _ -> assert false (* [t] must be an operator *)
```

Comment:
In the comment, you say that it returns 0 if it is not an operator but, here, you assert false.